sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -175,7 +176,7 @@ std::unique_ptr<RawResponse> CurlTransport::Send(Context const& context, Request
        getConnectionOpenIntent++)
   {
     performing = session->Perform(context);
-    if (performing != CURLE_UNSUPPORTED_PROTOCOL)
+    if (performing != CURLE_UNSUPPORTED_PROTOCOL && performing != CURLE_SEND_ERROR)
```

Comment:
`DefaultMaxOpenNewConnectionIntentsAllowed` is 10; it's totally possible to have more than 10 connections in the pool.

Another concern is that `CURLE_SEND_ERROR` is fairly generic.  If the connection came from the pool and you get a send error on the first try, retrying immediately is what you want, but if it's a subsequent send, or an error on a brand new connection, would it be more appropriate to go through back-off?

Reply:
That's a good point.
I have added more implementation details to the connection pool in this PR.  It now supports requesting for a new connections and forcing the pool to be cleaned up to produce a new fresh new connection.

If this pattern of asking for a new connection that fails to perform keeps repeating, after the 3rd time, the transport adapter will ask the pool to reset (clear connections to that host and with the same configuration) and open a new connection.

This resilient implementation will help to avoid asking the connection pool to always check if connections are valid before re-using it. Instead, the component requesting the connections can detect a fail pattern happening over and over and request the pool reset only if this ever happen